Algorithm Update + Multiobj integration test

gwenchee opened the pull request:
In this PR, I made the following changes 

1) updated ROLLO's algorithm to handle multi-objective optimization better, and updated the unit tests to reflect these changes. Changes reflected in: 
- `algorithm.py` 
- `test_algorithm.py`
2) updated selection operator to automatically assign the k parameter to be the population size. Previously, the user could input the value. This goes hand in hand with the new ROLLO algorithm update. Changes reflected in: 
- `input_validation.py`
- `toolbox_generator.py`
-  `test_toolbox_generator.py`
3) Moved all the initial tests into the unit_test directory and updated the CI workflow (#23). Changes reflected in: 
-  `.github/workflows/python-package-conda.yml`
4) introduced integration tests for single objective and multi objective test problems. Changes reflected in: 
- `test_ackley_integration.py`
- `test_binh_korn.py`
- `input_test_ackley.json`
- `input_test_ackley.py`
- `input_test_ackley_output.py`
- `input_test_binh.json`
- `input_test_binh.py`
- `input_test_binh_output.py`

pep8speaks commented:
Hello @gwenchee! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`rollo/constraints.py`](https://github.com/arfc/rollo/blob/237f63b2cbf7c42530c8be7d7829a3e56b0f828e/rollo/constraints.py):

> [Line 124:80](https://github.com/arfc/rollo/blob/237f63b2cbf7c42530c8be7d7829a3e56b0f828e/rollo/constraints.py#L124): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (122 > 79 characters)





* In the file [`rollo/input_validation.py`](https://github.com/arfc/rollo/blob/237f63b2cbf7c42530c8be7d7829a3e56b0f828e/rollo/input_validation.py):

> [Line 251:80](https://github.com/arfc/rollo/blob/237f63b2cbf7c42530c8be7d7829a3e56b0f828e/rollo/input_validation.py#L251): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 79 characters)
> [Line 254:80](https://github.com/arfc/rollo/blob/237f63b2cbf7c42530c8be7d7829a3e56b0f828e/rollo/input_validation.py#L254): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)

LukeSeifert reviewed (approved):
Everything looks good! I made a comment or two, but they are just suggestions and don't have to be implemented. 

I did notice that the functions in the testing modules don't have docstrings, so that may be something you want to add in a future version. I haven't worked much with tests though, so I'm not sure if adding documentation there is strictly necessary.

Go ahead and @ me when you're ready for me to merge.

gwenchee commented on tests/integration_tests/input_test_files/input_test_ackley_output.py:
```diff
@@ -0,0 +1,8 @@
+file = open("output.txt")
+contents = file.read()
+ackley = ast.literal_eval(contents)
+file.close()
```
Done :)

gwenchee commented:
Thanks for the review @LukeSeifert. I addressed your comments. I didn't include docstrings for my unit tests because they are a mirror that tests each function in the files in the rollo directory. So it is clear what each unit test is validating if you read the docstring of its corresponding function. For the integration tests, I included a description of the test at the top of the file. 

I'm ready to merge @LukeSeifert
